### PR TITLE
Change call to action in filter room when space selected

### LINKED ISSUE #3048: Room filter no results bad CTA in space mode when a space selected
image

Currently showing global action to create a new room (not a new room of that space)

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #3048

image

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt [hunk 1]
```diff
@@ -33,11 +34,21 @@ abstract class FilteredRoomFooterItem : VectorEpoxyModel<FilteredRoomFooterItem.
...
+    var inSpace: Boolean = false
+
     override fun bind(holder: Holder) {
         super.bind(holder)
         holder.createRoomButton.onClick { listener?.createRoom(currentFilter) }
         holder.createDirectChat.onClick { listener?.createDirectChat() }
         holder.openRoomDirectory.onClick { listener?.openRoomDirectory(currentFilter) }
+
+        holder.openRoomDirectory.setText(
+                if (inSpace) R.string.space_explore_activity_title else R.string.room_filtering_footer_open_room_directory
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt:47]: Q?: if we're always setting the text in the holder, should we also remove text from the layout? https://github.com/vector-im/element-android/pull/4124/files#diff-957e5d9987bce3ce857ba9f879945eab7131bc80e513432216e240e307aabf71R41

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt:47]: Your link is a self limk, you mean https://github.com/vector-im/element-android/blob/develop/vector/src/main/res/layout/item_room_filter_footer.xml#L41

Would be better to replace `android:text` by `tools:text` (and format)

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt:47]: argh it only works if the file is expanded :facepalm:

**Review by bmarty — APPROVED** [review]

File: vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/filtered/FilteredRoomFooterItem.kt:47]: I did the change in f6b672546791e56e9618e2b6ba195bd202b48320, @ouchadam, if it's OK for you, you can approve and auto-merge this PR

**Review by ouchadam — APPROVED** [review]

File: vector/src/main/res/layout/item_room_filter_footer.xml [hunk 2]
```diff
@@ -31,13 +32,14 @@
         android:layout_gravity="center"
         android:text="@string/room_filtering_footer_create_new_direct_message" />
 
+    <!-- Note: can also be used to open the SpaceExploreActivity-->
```
Comment (bmarty) [inline vector/src/main/res/layout/item_room_filter_footer.xml:35]: I prefer to add this comment instead of renaming id, funs, etc.

**Pull request merged**